File: sdk/Managed/src/Microsoft.WindowsAzure.MobileServices/Table/Sync/MobileServicePushStatus.cs
// ----------------------------------------------------------------------------
// Copyright (c) Microsoft Corporation. All rights reserved.
// ----------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Microsoft.WindowsAzure.MobileServices.Sync
{
    /// <summary>
    /// State in which push completed.
    /// </summary>
    public enum MobileServicePushStatus
    {
        /// <summary>
        /// All table operations in the push action were completed, possibly with errors.
        /// </summary>
        Complete = 0,

        /// <summary>
        /// Push was aborted due to network error.
        /// </summary>
        CancelledByNetworkError = 1,
        
        /// <summary>
        /// Push was aborted due to authentication error.
        /// </summary>
        CancelledByAuthenticationError = 2,
        
        /// <summary>
        /// Push was aborted due to error from sync store.
        /// </summary>
        CancelledBySyncStoreError = 3,
        
        /// <summary>
        /// Push was aborted due to cancellation.
        /// </summary>
        CancelledByToken = 4,

        /// <summary>
        /// Push was aborted by <see cref="IMobileServiceTableOperation"/>.
        /// </summary>
        CancelledByOperation = 5,

        /// <summary>
        /// Push failed due to an internal error.
        /// </summary>
        InternalError = Int32.MaxValue
    }
}

File: sdk/android/src/sdk/src/main/java/com/microsoft/windowsazure/mobileservices/table/sync/push/MobileServicePushStatus.java
/*
Copyright (c) Microsoft Open Technologies, Inc.
All Rights Reserved
Apache 2.0 License
 
   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 
See the Apache Version 2.0 License for specific language governing permissions and limitations under the License.
 */

/**
 * MobileServicePushStatus.java
 */
package com.microsoft.windowsazure.mobileservices.table.sync.push;

import android.annotation.SuppressLint;

import java.util.HashMap;
import java.util.Map;

/**
 * Enumeration for push sync status.
 */
@SuppressLint("UseSparseArrays")
public enum MobileServicePushStatus {

    /**
     * All table operations in the push action were completed, possibly with
     * errors.
     */
    Complete(0),

    /**
     * Push was aborted due to network error.
     */
    CancelledByNetworkError(1),

    /**
     * Push was aborted due to authentication error.
     */
    CancelledByAuthenticationError(2),

    /**
     * Push was aborted due to error from local store.
     */
    CancelledByLocalStoreError(3),

    /**
     * Push failed due to an internal error.
     */
    InternalError(Integer.MAX_VALUE);
    private static final Map<Integer, MobileServicePushStatus> mValuesMap;
    static {
        mValuesMap = new HashMap<Integer, MobileServicePushStatus>(4);
        mValuesMap.put(0, MobileServicePushStatus.Complete);
        mValuesMap.put(1, MobileServicePushStatus.CancelledByNetworkError);
        mValuesMap.put(2, MobileServicePushStatus.CancelledByAuthenticationError);
        mValuesMap.put(3, MobileServicePushStatus.CancelledByLocalStoreError);
        mValuesMap.put(Integer.MAX_VALUE, MobileServicePushStatus.InternalError);
    }
    private final int mValue;

    private MobileServicePushStatus(int value) {
        this.mValue = value;
    }

    /**
     * Return the MobileServicePushStatus with the provided int value
     *
     * @param value the int value
     * @return the matching MobileServicePushStatus
     */
    public static MobileServicePushStatus parse(int value) {
        return mValuesMap.get(value);
    }

    /**
     * Return the int value associated to the enum
     */
    public int getValue() {
        return this.mValue;
    }
}
